File: src/src_core/fmg/core/types/ClickResult.java
package fmg.core.types;

import fmg.core.mosaic.cells.BaseCell;

public class ClickResult extends ClickCellResult {

    public final BaseCell cellDown;
    public final boolean isLeft;
    public final boolean isDown;

    public ClickResult(BaseCell cellDown, boolean left, boolean down) {
        this.cellDown = cellDown;
        this.isLeft = left;
        this.isDown = down;
    }

    @Override
    protected String toStringInternal() {
        return "cellDown=" + cellDown
             + ", isLeft=" + isLeft
             + ", isDown=" + isDown
             + ", "  + super.toStringInternal();
    }

}

File: src/src_core/fmg/core/types/ClickResult.cs
using fmg.core.mosaic.cells;

namespace fmg.core.types.click {

    public class ClickResult : ClickCellResult {

        public ClickResult(BaseCell cellDown, bool left, bool down) {
            CellDown = cellDown;
            IsLeft = left;
            IsDown = down;
        }

        /// <summary> cell which was pressed (down click), but not the fact that the released (click released) </summary>
        public BaseCell CellDown { get; }

        /// <summary> its left click </summary>
        public bool IsLeft { get; }

        /// <summary> its down click </summary>
        public bool IsDown { get; }

    }

}
